src/test/java/org/cryptomator/ui/error/ErrorControllerTest.java

```diff
@@ -0,0 +1,148 @@
+package org.cryptomator.ui.error;
+
+import org.cryptomator.common.Environment;
+import org.cryptomator.common.ErrorCode;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.mockito.Mockito;
+
+import javafx.application.Application;
+import javafx.scene.Scene;
+import javafx.stage.Stage;
+
+import java.util.concurrent.ExecutorService;
+
+class ErrorControllerTest {
+	Application application;
+	String stackTrace;
+	ErrorCode errorCode;
+	Scene previousScene;
+	Stage window;
+	Environment environment;
+	ExecutorService executorService;
+	ErrorController errorController;
+
+	@BeforeEach
+	public void beforeEach() {
+		application = Mockito.mock(Application.class);
+		stackTrace = "This is a stackTrace mock";
+		errorCode = Mockito.mock(ErrorCode.class);
+		previousScene = Mockito.mock(Scene.class);
+		window = Mockito.mock(Stage.class);
+		environment = Mockito.mock(Environment.class);
+		executorService = Mockito.mock(ExecutorService.class);
+		errorController = new ErrorController(
+				application,
+				stackTrace,
+				errorCode,
+				previousScene,
+				window,
+				environment,
+				executorService);
+	}
+
+	private ErrorDiscussion createErrorDiscussion(String title, int upvoteCount, ErrorDiscussion.Answer answer){
+		ErrorDiscussion ed = new ErrorDiscussion();
+		ed.title =title;
+		ed.upvoteCount = upvoteCount;
+		ed.answer = answer;
+		return ed;
+	}
+
+	@ParameterizedTest
+	@CsvSource(textBlock = """
+		10, 5, -1 
+		8, 15, 1 
+		""")
+	public void testCompareUpvoteCount1(int leftUpvoteCount, int rightUpvoteCount, int expectedResult) {
+		var left = createErrorDiscussion("", leftUpvoteCount, null);
+		var right = createErrorDiscussion("", rightUpvoteCount, null);
+		int result = errorController.compareUpvoteCount(left, right);
+		Assertions.assertTrue(result*expectedResult>0);
```

Comment:
This is really hard to understand.

What you want to use is the signum function:

```suggestion
		Assertions.assertEquals(expected, Integer.signum(result));
```

Reply:
thanks, thats what i needed